src/Playwright.Tests.TestServer/SimpleServer.cs

```diff
@@ -62,53 +67,33 @@ public class SimpleServer
         public string CrossProcessPrefix { get; }
         public string EmptyPage { get; internal set; }
 
-        internal IList<string> GzipRoutes { get; }
-
-        public event EventHandler<RequestReceivedEventArgs> RequestReceived;
-
         public static SimpleServer Create(int port, string contentRoot) => new(port, contentRoot, isHttps: false);
 
         public static SimpleServer CreateHttps(int port, string contentRoot) => new(port, contentRoot, isHttps: true);
 
         public SimpleServer(int port, string contentRoot, bool isHttps)
         {
             Port = port;
-            if (isHttps)
-            {
-                Prefix = $"https://localhost:{port}";
-                CrossProcessPrefix = $"https://127.0.0.1:{port}";
-            }
-            else
-            {
-                Prefix = $"http://localhost:{port}";
-                CrossProcessPrefix = $"http://127.0.0.1:{port}";
-            }
+            var protocol = isHttps ? "https" : "http";
+            Prefix = $"{protocol}://localhost:{port}";
+            CrossProcessPrefix = $"{protocol}://127.0.0.1:{port}";
 
             EmptyPage = $"{Prefix}/empty.html";
 
-            _subscribers = new ConcurrentDictionary<string, Action<HttpContext>>();
             _requestWaits = new ConcurrentDictionary<string, Action<HttpContext>>();
             _waitForWebSocketConnectionRequestsWaits = new List<Action<HttpContext>>();
             _routes = new ConcurrentDictionary<string, RequestDelegate>();
             _auths = new ConcurrentDictionary<string, (string username, string password)>();
             _csp = new ConcurrentDictionary<string, string>();
-            GzipRoutes = new List<string>();
+            _gzipRoutes = new List<string>();
+            _contentRoot = contentRoot;
 
             _webHost = new WebHostBuilder()
-                .ConfigureAppConfiguration((context, builder) => builder
-                    .SetBasePath(context.HostingEnvironment.ContentRootPath)
-                    .AddEnvironmentVariables()
-                )
-                .UseWebRoot("assets")
```

Comment:
Mainly we don't serve assets anymore with the ASP .NET StaticFiles class, we do it now manually using the `ServeFile` method. So we can remove this.